tools/grep_invalid_enforce.sh

```diff
@@ -0,0 +1,128 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to grep invalid PADDLE checks by directory or file in the paddle/fluid/,
+#   the result show all invalid PADDLE checks in specified directory or file.
+
+# Usage: 
+#   - bash grep_invalid_enforce.sh [target directory or file] (run in tools directory)
+#       - The default check path is paddle/fluid/operators
+
+# Result Examples:
+# 1. grep invalid PADDLE checks in directory
+
+#     - Command: /work/paddle/tools {develop} bash grep_invalid_enforce.sh ../paddle/fluid/imperative 
+#     - Results:
+#         - paddle/fluid/imperative/gradient_accumulator.cc
+#         PADDLE_ENFORCE_EQ(dst_tensor->numel() == numel, true,
+#                             "dst_numel %d vs. src_numel %d", dst_tensor->numel(),
+#                             numel);
+
+#         - paddle/fluid/imperative/nccl_context.cc
+#         PADDLE_ENFORCE_EQ(addr.size(), 2UL,
+#                             "The endpoint should contain host and port: %s", ep);
+#         PADDLE_THROW("create server fd failed");
+#         PADDLE_THROW("set socket opt failed");
+#         PADDLE_THROW("binding failed on ep: %s", ep);
+#         PADDLE_THROW("listen on server fd failed");
+#         PADDLE_THROW("accept the new socket fd failed");
+#         PADDLE_THROW("reading the ncclUniqueId from socket failed");
+#         PADDLE_ENFORCE_EQ(addr.size(), 2UL,
+#                             "The endpoint should contain host and port: %s", ep);
+#         PADDLE_THROW("create socket failed");
+#         PADDLE_THROW("invalied address: %s", ep);
+
+#         - paddle/fluid/imperative/jit/program_desc_tracer.cc
+#         PADDLE_ENFORCE_NOT_NULL(new_var);
+#         PADDLE_ENFORCE_EQ(inner_var.IsInitialized(), true);
+#         PADDLE_THROW("Not support variable type %s",
+#                         framework::ToTypeName(inner_var.Type()));
+
+# 2. grep invalid PADDLE checks in file
+
+#     - Command: /work/paddle/tools {develop} bash grep_invalid_enforce.sh ../paddle/fluid/pybind/reader_py.cc
+#     - Results:
+#         - paddle/fluid/pybind/reader_py.cc
+#         PADDLE_THROW(
+#                     "Place cannot be CUDAPlace when use_double_buffer is False");
+#         PADDLE_ENFORCE_NOT_NULL(exceptions_[i]);
+#         PADDLE_ENFORCE_EQ(status, Status::kException);
+#         PADDLE_ENFORCE_EQ(status, Status::kSuccess);    
+
+ROOT_DIR=../paddle/fluid/operators
+
+if [ "$1" != "" ]; then
+    ROOT_DIR=$1
+fi
+
+white_list_str="\
+    layer_norm_op.cc \
+    box_clip_op.cc \
+    box_clip_op.h \
+    random_crop_op.h \
+    elementwise_op_function.cu.h \
+    fused_elemwise_activation_op.cc \
+    auc_op.cu"
+
+function enforce_scan(){
```

Comment:
Not important.
I saw this function in several scripts, should we reuse it instead of making code copy?

Reply:
`enforce_scan` in `grep_invalid_enforce.sh` is not same as this function in other shell, but `enforce_scan` in other three shell scripts are same, we can reuse them later.

Reply:
For example, the `regex pattern` for valid and invalid are the same, we can reuse them.

Reply:
done, reuse public function, polish function name